packages/examples/graphiql-webpack/webpack.config.js

```diff
@@ -0,0 +1,71 @@
+const webpack = require("webpack");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const { TsConfigPathsPlugin } = require("awesome-typescript-loader");
+const path = require("path");
+const isDev = process.env.NODE_ENV === "development";
+
+module.exports = {
+  entry: isDev
+    ? [
+        "react-hot-loader/patch", // activate HMR for React
+        "webpack-dev-server/client?http://localhost:8080", // bundle the client for webpack-dev-server and connect to the provided endpoint
+        "webpack/hot/only-dev-server", // bundle the client for hot reloading, only- means to only hot reload for successful updates
+        "./index.tsx" // the entry point of our app
+      ]
+    : "./index.tsx",
+  context: path.resolve(__dirname, "./src"),
+  mode: "development",
+  devtool: "inline-source-map",
+  externals: {
+    react: "React",
+    "react-dom": "ReactDOM"
+  },
+  performance: {
+    hints: false
+  },
+  module: {
+    rules: [
+      {
+        test: /\.html$/,
+        use: ["file?name=[name].[ext]"]
+      },
+      {
+        type: "javascript/auto",
+        test: /\.mjs$/,
+        use: [],
+        include: /node_modules/
+      },
+      {
+        test: /\.(ts|tsx)$/,
+        use: [{ loader: "awesome-typescript-loader" }]
+      },
+      {
+        test: /\.css$/,
+        use: ["style-loader", "css-loader"]
+      },
+      {
+        test: /\.svg$/,
+        use: [{ loader: "svg-inline-loader" }]
+      }
+    ]
+  },
+  resolve: {
+    extensions: [".js", ".json", ".ts", ".tsx", ".jsx", ".css", ".mjs"],
+    plugins: [new TsConfigPathsPlugin()]
+  },
+  plugins: [
+    new HtmlWebpackPlugin({ template: "index.html.ejs" }),
+    new webpack.ContextReplacementPlugin(/graphql-language-service-interface[\\/]dist$/, new RegExp(`^\\./.*\\.js$`)),
+    new webpack.ContextReplacementPlugin(/graphql-language-service-utils[\\/]dist$/, new RegExp(`^\\./.*\\.js$`)),
+    new webpack.ContextReplacementPlugin(/graphql-language-service-parser[\\/]dist$/, new RegExp(`^\\./.*\\.js$`))
```

Comment:
Just stumbled upon this PR wondering why there were a bunch of warnings emitted with the latest version.  These lines fix it.  Curious why this is necessary and the resolve.extension order wasn't respected.

Reply:
@gpoitch yes this is exactly the issue we're trying to solve in this PR actually. i've introduced this working webpack config, and once i have the webpack example working all the way to netlify, i will disable these plugin invocations until it is working transparently upstream.

we believe we just have to import dynamically using `require`, or a dynamic import with an explicit extension for webpack to be happy, at least we think that might help. ideally i'd like to be doing things like this:

```js
await import(`graphiql-plugin-${plugin}.js`)
```
for now still using `require` and just iterating slowly til we get there. i guess we'll be breaking people's bundler configs no matter what

Reply:
https://github.com/graphql/graphiql/pull/973/files#diff-f9580b2b8f2bad829ef1defe703f8e18L162 specifically i beleive its this line, and the other `require` a few lines below it. they were introduced in the 2.x version LSP earlier this year, but we just released the LSP again after a year of no releases so yeah

Reply:
hey so, good news @gpoitch, i got this working locally, about to push a version with a working netlify webpack build that doesn't require the special config.

Reply:
```js
const webpack = require("webpack");
const HtmlWebpackPlugin = require("html-webpack-plugin");
const path = require("path");
const isDev = process.env.NODE_ENV === "development";

module.exports = {
  entry: isDev
    ? [
        "react-hot-loader/patch", // activate HMR for React
        "webpack-dev-server/client?http://localhost:8080", // bundle the client for webpack-dev-server and connect to the provided endpoint
        "webpack/hot/only-dev-server", // bundle the client for hot reloading, only- means to only hot reload for successful updates
        "./index.jsx" // the entry point of our app
      ]
    : "./index.jsx",
  context: path.resolve(__dirname, "./src"),
  mode: "development",
  devtool: "inline-source-map",
  externals: {
    react: "React",
    "react-dom": "ReactDOM"
  },
  performance: {
    hints: false
  },
  module: {
    rules: [
      {
        test: /\.html$/,
        use: ["file?name=[name].[ext]"]
      },
      {
        type: "javascript/auto",
        test: /\.mjs$/,
        use: [],
        include: /node_modules/
      },
      {
        test: /\.(js|jsx)$/,
        use: [{ loader: "babel-loader", options: { presets: ["@babel/preset-env", "@babel/preset-react"]} }]
      },
      {
        test: /\.css$/,
        use: ["style-loader", "css-loader"]
      },
      {
        test: /\.svg$/,
        use: [{ loader: "svg-inline-loader" }]
      }
    ]
  },
  resolve: {
    extensions: [".js", ".json", ".jsx", ".css", ".mjs"],
  },
  plugins: [
    new HtmlWebpackPlugin({ template: "index.html.ejs" }),
  ],
  devServer: {
    hot: true,
    // bypass simple localhost CORS restrictions by setting
    // these to 127.0.0.1 in /etc/hosts
    allowedHosts: ["example.com"]
  },
  node: {
    fs: "empty"
  }
};

```

Reply:
this works as it should! :D

Reply:
@gpoitch the example is ready now, seems to be working, and now we have tree shaking too!